htdocs/core/modules/modTakePos.class.php

```diff
@@ -323,6 +323,24 @@ public function init($options = '')
 			}
 		}
 
+		//Create cash account if not exists
+		if (!getDolGlobalInt('CASHDESK_ID_BANKACCOUNT_CASH1')) {
+			require_once DOL_DOCUMENT_ROOT.'/compta/bank/class/account.class.php';
+			$cashaccount = new Account($db);
+			$cashaccount->ref = "CASH-POS";
+			$cashaccount->label = $langs->trans("DefaultCashPOSLabel");
+			$cashaccount->courant = 2;
+			$cashaccount->country_id = $mysoc->country_id ? $mysoc->country_id : 1;
+			$cashaccount->date_solde = dol_now();
+			$result = $cashaccount->create($user);
```

Comment:
Before creating the account, can you add a check before with a fetch() and create only if it does not exists. Like done for the default thirdparty.